### ISSUE TITLE
Incorrect nullable warning with constructor field initialization one function deep

### ISSUE THREAD

**Issue by myblindy** [dotnet/roslyn#46040]
**Version Used**: 
.Net 5.0 preview 6

**Steps to Reproduce**:

```c#
        class C
        {
            object a;

            C()
            {
                Init();
            }

            private void Init()
            {
                a = new object();
            }
        }
```

**Expected Behavior**:

I would expect the compiler to realize the function is always initialized. The pattern here is important because `a` is actually a Vulcan vertex buffer handle that resizes as needed, and the function handles the disposing of the previous buffer and allocating a new one, and if I were to add the allocation to the constructor too it would be duplicated code for the sake of pleasing the compiler.

**Actual Behavior**:

The code generates a nullable warning for `C.a`:
```
Warning	CS8618	Non-nullable field 'a' is uninitialized. Consider declaring the field as nullable.
```

**Comment by RikkiGibson** [comment]
Duplicate of #32358.

Currently the solution for this is to apply `[MemberNotNull(nameof(a))]` to `Init()`.
